### PR TITLE
Factor panel actions/store/lock out of ui actions/store/lock

### PR DESCRIPTION
**Pull request by iwehrman**
This PR factors a panel action module, store and lock out from the existing UI action module, store and lock. This reduces contention for the UI lock somewhat at startup time, and also arguably also makes this  code a little easier to understand. In some cases it wasn't clear what state/action should go where, but the rule of thumb I tried to use was: if it has to do with the _window_ put it in the UI store, and if it has to do with a _panel_ put it in the panel store. (We should probably rename the UI store to something more appropriate, but I don't currently have a suggestion, so I left it alone.) This seems to shave ~100ms off startup time on my MBP due to reduced lock contention.

### CODE REVIEW

**Comment by volfied** [comment]
Feels like marquee related information should be in UI store. Or maybe I don't understand the logic enough yet.

**Comment by iwehrman** [comment]
The hope was to move more of the app/feature-specific stuff out of the UI store. So, I disagree!

File: src/js/actions/ui.js [hunk 1]
```diff
@@ -290,22 +290,22 @@ define(function (require, exports) {
      * @return {Promise}
      */
     var updatePanelSizes = function (sizes) {
-        return this.dispatchAsync(events.ui.PANELS_RESIZED, sizes)
+        var transformPromise = this.transfer(updateTransform),
+            dispatchPromise = this.dispatchAsync(events.panel.PANELS_RESIZED, sizes)
             .bind(this)
```
Comment (volfied) [inline src/js/actions/ui.js]: This needs to be indented.

**Comment by volfied** [comment]
You forgot some `events.ui.TOGGLE_OVERLAYS` (at least one in transform.js#beforeStartup)

**Comment by volfied** [comment]
I pushed a commit myself fixing this, since I'm likely to merge it after giving it one more look.

**Comment by volfied** [comment]
We can probably sit and argue about the dividing line all day, but since this helps startup speed and doesn't hurt anything, I'm merging it. And will make another PR for more lock cleanup.

**Pull request merged**